streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamPartitionAssignor.java

```diff
@@ -361,8 +388,10 @@ public Subscription subscription(Set<String> topics) {
 
         // assign tasks to clients
         states = TaskAssignor.assign(states, partitionsForTask.keySet(), numStandbyReplicas);
-        Map<String, Assignment> assignment = new HashMap<>();
 
+        final List<AssignmentSupplier> assignmentSuppliers = new ArrayList<>();
+
+        final Map<HostInfo, Set<TopicPartition>> endPointMap = new HashMap<>();
```

Comment:
One orthogonal thought: if we enforce the host info config, then we can get rid of the ProcessorID as UUID but just use the HostInfo, which will be more helpful for debugging / etc.